SmartDeviceLink/SDLAudioStreamManager.h

```diff
@@ -0,0 +1,60 @@
+//
+//  SDLBinaryAudioManager.h
+//  SmartDeviceLink-Example
+//
+//  Created by Joel Fischer on 10/24/17.
+//  Copyright © 2017 smartdevicelink. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@class SDLAudioFile;
+@class SDLManager;
+@class SDLStreamingMediaLifecycleManager;
+@protocol SDLStreamingAudioManagerType;
+@protocol SDLAudioStreamManagerDelegate;
+
+NS_ASSUME_NONNULL_BEGIN
+
+extern NSString *const SDLErrorDomainAudioStreamManager;
+
+typedef NS_ENUM(NSInteger, SDLAudioStreamManagerError) {
+    SDLAudioStreamManagerErrorNotConnected = -1,
+    SDLAudioStreamManagerErrorNoQueuedAudio = -2
+};
+
+@interface SDLAudioStreamManager : NSObject
+
+@property (weak, nonatomic) id<SDLAudioStreamManagerDelegate> delegate;
+
+@property (assign, nonatomic, readonly, getter=isPlaying) BOOL playing;
+@property (copy, nonatomic, readonly) NSArray<SDLAudioFile *> *queue;
+
+- (instancetype)init NS_UNAVAILABLE;
+
+- (instancetype)initWithManager:(id<SDLStreamingAudioManagerType>)streamManager NS_DESIGNATED_INITIALIZER;
+
+/**
+ Push a new file URL onto the queue after converting it into the correct PCM format for streaming binary data.
+
+ @note This happens on a global background thread and will provide an error callback using the delegate if the conversion fails.
+
+ @param fileURL File URL to convert
+ */
+- (void)pushWithFileURL:(NSURL *)fileURL;
+
+/**
+ Play the next item in the queue. If an item is currently playing, it will continue playing and this item will begin playing after it is completed. If all items are playing via `playAll`, this will do nothing.
+
```

Comment:
I can't find a "playAll" setting.

Reply:
There is no `playAll` setting, and was none in the agreed upon interface. If you wanted to do something like this, right now you'd have to be the delegate and play the next one when the previous one finishes.

Reply:
It says "playAll" in the comment:
> + Play the next item in the queue. If an item is currently playing, it will continue playing and this item will begin playing after it is completed. If all items are playing via `playAll`, this will do nothing.

Reply:
Ooops 😅